data_structure/linked_list.cpp

```diff
@@ -1,57 +1,71 @@
 #include <iostream>
+template  <class X>
+class node {
+public:
+    X val;
+    node<X> *next;
```

Comment:
adding a constructor and destructor would help - otherwise, this is a plain  structure.
` node(X value) (val: value) {;}`
should do.

A recursive destructor - to ensure clean up of the `*next` on exit